src/email.rs

```diff
@@ -42,6 +42,7 @@ pub fn request(app: &AppConfig, email_addr: EmailAddress, client_id: &str, nonce
     // Store data for this request in Redis, to reference when user uses
     // the generated link.
     try!(app.store.store_session(&session, &[
+        ("type", "email"),
```

Comment:
I think it might make more sense to hash the type into the session key, instead. What do you think?